text/0025-core-compute-resources-as-plugins.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: Core Compute Resources as Plugins
+- Start Date: 2016-10-21
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Core Compute Resources should move to plugins
+
+# Motivation
+[motivation]: #motivation
+
+Core has been set up to allow compute resources as plugins for some while, and
+we already have several examples of this in use (Xen, Azure, DigitalOcean,
+etc). This has brought benefits:
+
+* able to release independatly of Core
+* make use of the modular Fog gems that are now available
+* separate logging of issues
+* dedicated maintainers who know the compute resource well
```

Comment:
That's true. But at least for VMWare I think a proper maintainer is missing in core. I can't tell about other compute resources. If the plugin API in core is missing features, we can add them so that no hacks are necessary. I'd rather have a plugin in core that receives attention than a compute resource in core that is stable but doesn't support all required features. Extracting a compute resource to a plugin can actually also lead to besser code quality when the chance is used to add tests or refactor things.